### ISSUE TITLE
A flat vinyl print passes every native cross-spectrum cue; only the third-party PAD model denies it

### ISSUE THREAD

**Issue by archledger** [archledger/irlume#235]
Measured on the Zenbook, 2026-08-02 evening (setup: ASUS FHD module IR pin, GREY8 640x400, daemon built from main at 1289ae4, `IRLUME_LOG=debug`, dark room). A vinyl print of the enrolled user's face (a graduation banner) was held facing the camera and `irlume identify` was run four times. All four attempts produced this verdict from the native gate:

```
liveness(cross-spectrum): Live (live: face in RGB+IR, co-located, frontal, IR-reflective, 3D)
```

The denial came only from the third-party PAD stage each time:

```
thirdparty-pad('flir'): p_fake 0.993 >= 0.90; downgrading Live to Spoof
```

The four cross-spectrum readings for the print:

| attempt | ir_bright | centre/edge | glint | face_frac | ir_clipped | flir p_fake |
|---|---|---|---|---|---|---|
| 1 | 116 | 1.12 | 130 | 0.315 | 1.9% | 0.993 |
| 2 | 76 | 1.12 | 84 | 0.302 | 0.2% | 1.000 |
| 3 | 98 | 1.17 | 107 | 0.356 | 0.6% | 0.999 |
| 4 | 70 | 1.17 | 84 | 0.187 | 0.0% | 0.995 |

What each native cue did against the print:

- **IR reflectivity:** the vinyl reproduces the whole face in IR at full contrast (burst frames on #222), so "a real face reflects 850nm" holds for vinyl too.
- **Centre/edge ratio (the 3D cue):** 1.12 to 1.17, above the 1.03 floor in every attempt. A flat print defeats the floor as tuned.
- **Glint:** 84 to 130 against the pegged 255.00 of nine out of ten genuine logins tonight. The cue separated print from cornea in this data, but the earlier glossy-print result (69/70) shows a surface with the right gloss can peg it, so it is not a floor to lean on.
- **Co-location, frontality:** both satisfied by a print held square to the camera.

Why the floor cannot simply be raised: #221's measurements put a saturated genuine capture at ratio 1.11, below the print's 1.12, while unclipped genuine captures read 1.35 to 1.54. The genuine and spoof distributions only separate after clipped frames stop being gated. Sequence that follows: land least-clipped-frame selection (#221), re-measure the genuine floor, then retune the 1.03 threshold with the print data as the attack reference.

One earlier attempt in the same sitting failed differently and is worth recording as defense-in-depth behaving as designed: with the print poorly lit, the RGB stage found no face (verdict Uncertain, never reached matching) and flir abstained at p_fake 0.679, between the measured genuine ceiling and the 0.90 deny threshold.

Layer status: the outcome was deny in all four attempts, so nothing here is an open hole today. The finding is that the native cross-spectrum gate contributed nothing against this attack class, and the third-party model is currently a single point of failure for flat-print PAD. Related: #174 (distance behaviour of the same ratio), #221 (saturation compressing it), #222 (glint peak semantics), #101 (mean_step corpus).

Not tested: the same print under daylight ambient, curved presentation (wrapping the print to fake depth), other modules. One subject, one room.

**Comment by archledger** [comment]
Measured tonight, and the conclusion is that this cue cannot be fixed by tuning it.

An 84-presentation corpus was captured with `padcapture` (ASUS module, dark room, one subject) to see whether the 1.03 floor could rise far enough to reject the vinyl print. Held square to the camera it looked possible:

| presentation | n | ratio | called Live at the 1.03 floor |
|---|---|---|---|
| vinyl print, normal distance | 12 | 1.02 to 1.16 | 10 |
| vinyl print, close | 12 | 1.12 to 1.21 | 12 |
| live face, five conditions | 60 | 1.26 to 1.49 | 58 |

22 of 24 print presentations accepted. The print's ceiling was 1.21 and the lowest genuine capture that passed liveness read 1.31, so a floor near 1.25 appeared to separate them, and PR #239 did exactly that.

Then the banner was presented ANGLED rather than square: 20 presentations, reaching **1.51**, four of them accepted even against the proposed 1.25 floor. Blocking those needs a floor above 1.51, which rejects all 58 genuine captures. #239 is closed.

This is the 2026-06-30 selftest again, which measured the same banner at 1.02 to 1.58 across 70 varied presentations and wrote that no floor value accepts the live user and rejects the banner. That document was in the repository the whole time and I did not read it before building the corpus.

**Where that leaves the issue.** The built-in gate cannot solve this by threshold choice, on this hardware. The cue is a brightness ratio on a 2D IR sensor, and an attacker who tilts a life-size print produces the same falloff a face does. Two directions that are not threshold tuning:

1. Make the third-party PAD cue a default rather than an opt-in. It denied the print at p_fake 0.999 and above on unclipped frames (#237 covers the clipped case, now fixed). This is the only measured defence that works, and today a default install does not have it.
2. A cue that is not a brightness ratio: #25's landmark-anchored local relief, or the bright-pupil separation #222 explored and found unavailable on all three supported modules.

Corpus and analysis in PR #240.

**Comment by archledger** [comment]
Re-measured with the print in hand this morning (2026-08-04, Zenbook ASUS FHD module, daemon at 8ea6de2, room lit): 8 genuine presentations plus 24 attack presentations across three geometries, through the real gate.

| presentation | n | native verdict | centre/edge | glint | ir brightness |
|---|---|---|---|---|---|
| genuine face | 8 | 8 Live | 1.44-1.54 | 255 (pegged) | 72.7-87.1 |
| print, flat and square | 8 | 8 Live (breach) | 1.04-1.12 | 105-158 | 86.2-135.4 |
| print, curved toward the camera | 8 | 8 Live (breach) | 1.15-1.20 | 87-137 | 81.7-127.7 |
| print, angled 20-30° | 8 | 8 Live (breach) | 1.10-1.15 | 69-93 | 58.6-70.9 |

The issue's finding reproduces exactly: the native cross-spectrum gate accepted every one of the 24 attack presentations, and the only thing standing between this print and a credential is the opt-in third-party model.

## The curved case, which this issue listed as untested

Curving the print raised its ratio from 1.04-1.12 to 1.15-1.20. That is the important part: **the attacker steers this cue**. It is not a fixed spoof distribution that a well-chosen floor separates from genuine; it moves in the defender's direction as a function of how the attacker bends the paper, and this was a hand-held bow, not a mould.

## What today's separation does NOT license

In this session the populations look cleanly separable (genuine 1.44-1.54 against attacks 1.04-1.20, a 0.25 gap), and it would be easy to read that as "raise the floor to 1.3 and be done". The repository's own record refutes that, and I checked it before writing this: on 2026-08-02 the same banner at varied angles reached **1.51**, inside today's genuine range, at brightness 97 with zero clipping, so it was not an exposure artifact. Two sessions, same camera, same subject, same instrument; the attack's reachable ceiling moved by 0.3 between them. A floor fitted to today would have been defeated by a presentation already recorded two days ago. Today's separation is a property of this morning's conditions, not of the cue.

That is the same lesson as #239 (closed unmerged for exactly this), now with a second independent confirmation, and it applies to the glint cue as well: glint separated perfectly today (genuine pegged 255 against attacks 69-158), yet the June self-test recorded a glossy print that pegged it.

## Where this leaves the issue

The native gate has no cue that is both attacker-proof and threshold-able on this hardware. Raising ratios is closed as a direction. What remains is a cue whose physics a flat or bent sheet cannot reproduce, which is #25's landmark-anchored relief: nose brighter than cheek brighter than eye socket, measured at known facial positions rather than as one global centre-versus-edge number. Raw 30-frame IR bursts of the curved print and the genuine face, captured minutes apart in the same light, are stored for that work (`~/irlume-research/2026-08-04-print-corpus/burst-*`, PGM plus per-frame means), so #25 can be prototyped offline against a real attack instrument instead of against theory.

## On the models suggested today

Three repositories were raised as candidates. None addresses this gap, and I checked their provenance rather than their accuracy claims:

- **facenox/face-antispoof-onnx** repackages MiniVision's MiniFASNetV2-SE. Apache-2.0 covers the code; the README names CelebA-Spoof only as an evaluation set and does not state what the weights were trained on. CelebA-Spoof is the non-commercial dataset already blocking irlume's commercialization path, so this would add a second model with the same ADR-0001 criteria 2-3 failure as `flir`, not resolve one. It is also 128x128 **RGB**, and this attack defeats the IR cues specifically.
- **opencv_zoo SFace** is a recognition model (MobileFaceNet + SFace loss), an alternative embedder, not an anti-spoof cue; the zoo page does not name its training data either.
- **opencv_zoo facial expression recognition** classifies seven emotions from RAF-DB. Expression *change* is a challenge-response idea, which ADR-0002 already covers, and #101 measured that a held print performs motion-based challenges convincingly.

Not tested today: daylight ambient, a print on a curved rigid mould rather than hand-bowed, other camera modules. One subject, one room, as before.

**Comment by archledger** [comment]
A finding from the #25 occluder session this afternoon that belongs here too, because it concerns the shipped gate rather than a candidate cue.

**A black cotton sock held over the print's chin moves the shipped centre/edge ratio into the genuine band.** Measured on the same banner, same camera, same afternoon:

| presentation | centre/edge ratio | accepted by the gate |
|---|---|---|
| bare print | 1.08-1.15 | 6/6 |
| print + black polyester | 0.99-1.11 | 4/6 |
| print + black cotton | 1.31-1.53 | 6/6 |
| genuine face, same day | 1.44-1.54 | — |

No verdict changes today, because the bare print is already accepted 6 of 6, so this is not a new hole. What it changes is the value of the one remedy this issue's thread keeps circling back to.

The 2026-08-02 corpus already showed tilt reaching 1.51 against a genuine 1.44-1.54, which is what closed the 1.25 floor proposal. This is stronger: an attacker with a household item can place the print's ratio *inside* the genuine band rather than merely at its edge, and can do it while the print is flat and square. Any floor set anywhere within the genuine distribution is reachable that way. Raising `MIN_CENTER_EDGE_RATIO` is not a partial mitigation against a prepared attacker; it is no mitigation.

Material matters and colour does not predict it, which is worth recording for anyone reproducing this: the black polyester tried first *reflects* 850nm and made the chin brighter than the forehead, a worse attack than no occluder at all. Cotton absorbs.

Corpora and full numbers in `docs/pad-results/2026-08-04-occluder.md` (PR #272).

**Comment by archledger** [comment]
Measured today with the model enabled, which the occluder work had not been: **the occluder that defeats every native cue does not move flir.**

| presentation | native verdict | centre/edge | flir p_fake |
|---|---|---|---|
| bare print | Live | 1.06, 1.06 | 0.988, 1.000 |
| print + cotton over the chin | Live | 1.32, 1.42, 1.44, 1.33 | 0.999, 0.999, 0.998, 0.999 |

Six runs through the daemon's real authentication path, all unclipped. The occluder carries the native centre/edge cue from 1.06 into 1.32-1.44, inside the committed genuine population (1.26-1.49), and the native gate returns Live on all six. The trained cue is unaffected.

This sharpens what this issue records rather than changing it. The native gate contributed nothing against the bare print; it now also contributes nothing against an enhanced one, while the enhancement is a household sock. The single point of failure this issue named is still a single point of failure, but its margin survived the one attack that beats everything else here.

Scope worth keeping attached to that: every frame was unclipped, and clipping is exactly this model's measured failure mode (p_fake 0.749 at 24.8% of the face at the ceiling, #237, which auto-exposure produces unaided after a daemon restart at close range). One model, one print, one subject. PR #273.

**Comment by archledger** [comment]
Closing as addressed — the finding became the design.

**Resolution trail:**

- **PR #516 (ADR-0013): both PAD models ship default-on.** The FLIR IR PAD that was 'third-party' when this issue was filed is now a shipped models-v1 asset (sha-pinned, startup-verified, kill-switch `IRLUME_PAD_IR=0` disclosed), wired deny-only into the cross-spectrum path, the RGB-only tier, and the dark path. The ViT RGB PAD (`liveness_vit.onnx`, m96/0.55/5-vote) adds the same print-species coverage on the RGB side.
- **Fleet validation caught this exact species 100%:** the vinyl banner was caught in both #516 qualification sessions on both dual-camera hosts (ASUS 0.653-0.742 pre-tune → 100% post-tune; minihost banner caught after the fleet threshold tune), with 0/180 genuine false-fires. The occluder data in the last comment (sock-over-chin defeats every native cue, FLIR stays 0.998+) is the standing proof the defense is not native-cue-dependent.
- **ADR-0016 codifies it:** the life-size print species is explicitly PAD + IR physics + per-user floor's job — no match threshold can catch it (the banner scores IR cosine 0.650, above even the 0.635 dark bar), so 'only the PAD denies the print' is the intended architecture, not an open defect.
- **PR #520 (ADR-0015)** removed the BYOM/third-party lane entirely; the shipped FLIR+ViT are the product path.

Residual, disclosed and intentional: with the PAD kill switch engaged, the print species has no native backstop — that is the documented cost of `IRLUME_PAD_IR=0`, chosen knowingly (ADR-0013).
